### PR TITLE
DO-987 updated the samza-deployment script with refactored scripts

### PR DESCRIPTION
**Pull request by Kaali09**
Updating the samza-jobs deployment script to achieve the below requirements:

Refactor plan :
1. The user should be able to deploy the single or multiple samza application.
2. Script should not look for samza application version and it should not care about what version of samza jobs is being to deploy.
3. For each deployment , the mentioned samza apps tar files should be removed and new tar files will be copied to the server.

Deployment steps will be:
1. Take the user input on which samza apps needs to be deployed.
2. Stop the Mentioned apps.
3. Remove the old tar files from 2 directories related to stopped apps.
4. copy the new tar files into the servers.
5. Create the directory to extract the new apps.
6. Find and replace the variables for newly copied apps.
7. start the new samza apps

### CODE REVIEW

File: ansible/roles/samza-jobs-additional-config/tasks/main.yml [hunk 1]
```diff
@@ -1,12 +1,29 @@
 ---
 - name: Create directory for additional config
-  file: path={{object_denormalization_additional_config_dir}} owner=hduser group=hadoop recurse=yes state=directory
+  file:
+    path: "{{ object_denormalization_additional_config_dir }}"
```
Comment (anandp504) [inline ansible/roles/samza-jobs-additional-config/tasks/main.yml]: Object Denormalization job is no longer used and can be removed.

File: ansible/roles/samza-jobs-additional-config/tasks/main.yml [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 - name: Copy Object denormalization additional config
-  template: src=object-denormalization-additional-config.json dest={{object_denormalization_additional_config}} owner=hduser group=hadoop
+  template:
+    src: object-denormalization-additional-config.json
+    dest: "{{ object_denormalization_additional_config }}"
+    owner: hduser
+    group: hadoop
 
 - name: Copy Primary es router additional config
-  template: src=es-router-additional-config.json dest={{es_router_additional_config}} owner=hduser group=hadoop
```
Comment (anandp504) [inline ansible/roles/samza-jobs-additional-config/tasks/main.yml]: es-router job is no longer used as well and hence can be removed.

File: ansible/samza_deploy.yml [hunk 3]
```diff
@@ -1,33 +1,32 @@
 ---
-- name: "Start Nodemanager on Slaves"
+- name: "Start Nodemanagerif on Slaves if stopped and Copy Samza jobs additional configuration to slaves"
```
Comment (anandp504) [inline ansible/samza_deploy.yml]: Please fix the description.

File: ansible/roles/samza-jobs/defaults/main.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-samza_jobs_dir: /home/hduser/samza-jobs/{{env}}
-job_status_file: /home/hduser/samza-jobs/{{env}}/extract/job_status
+samza_jobs_dir: /home/hduser/samza-jobs/{{ env }}
+job_status_file: /home/hduser/samza-jobs/{{ env }}/extract/job_status
 yarn_path: /usr/local/hadoop/bin
-object_denormalization_additional_config_dir: /etc/samza-jobs/{{env}}
-object_denormalization_additional_config: "{{object_denormalization_additional_config_dir}}/object-denormalization-additional-config.json"
-es_router_additional_config: "{{object_denormalization_additional_config_dir}}/es-router-additional-config.json"
-es_router_additional_secondary_config: "{{object_denormalization_additional_config_dir}}/es-router-additional-secondary-config.json"
+object_denormalization_additional_config_dir: /etc/samza-jobs/{{ env }}
```
Comment (anandp504) [inline ansible/roles/samza-jobs/defaults/main.yml]: These additional configurations are not required.

File: ansible/roles/samza-jobs-additional-config/tasks/main.yml (hunk 1 above)
Comment (Kaali09) [reply ansible/roles/samza-jobs-additional-config/tasks/main.yml]: removed the role as requested.

**Pull request merged**